### PR TITLE
fix(sqlness): sqlness isolation

### LINKED ISSUE #4779: sqlness test suite lacks isolation
### What type of bug is this?

User Experience

### What subsystems are affected?

Other

### Minimal reproduce step

Inside project root:

```bash
mkdir /tmp/{greptime,metasrv,flownode,demo,export}
sudo chown -R root:root /tmp/{greptime,metasrv,flownode,demo,export}
cargo sqlness
```

### What did you expect to see?

sqlness test cases all pass since we expected that each test uses their own directory like `/tmp/sqlness[...]/

### What did you see instead?

```
metasrv log file at /tmp/sqlnessVeVVks/greptime-sqlness-metasrv.log
Generating metasrv config file in /tmp/sqlnessVeVVks/greptimedb-metasrv/metasrv-1727471462905.toml, full content:
[wal]

provider = "raft_engine"


thread 'main_spawn' panicked at /home/lei/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-appender-0.2.3/src/rolling.rs:154:14:
initializing rolling file appender failed: InitError { context: "failed to create log directory", source: Os { code: 13, kind: PermissionDenied, message: "Permission denied" } }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at src/cmd/src/bin/greptime.rs:86:10:
Couldn't join on the associated thread: Any { .. }
thread 'main' panicked at tests/runner/src/env.rs:289:13:
metasrv doesn't up in 10 seconds, quit.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### What operating system did you use?

NA

### What version of GreptimeDB did you use?

latest

### Relevant log output and stack trace

_No response_

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
- fixes #4779

## What's changed and what's your intention?
Fix the isolation issue of sqlness test suite.

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by sunng87 — APPROVED** [review]
Awesome

LGTM

File: tests/cases/standalone/common/copy/copy_database_from_fs_parquet.result [hunk 1]
```diff
@@ -6,11 +6,11 @@ INSERT INTO demo(host, cpu, memory, ts) values ('host1', 66.6, 1024, 16552765570
 
 Affected Rows: 2
 
-COPY DATABASE public TO '/tmp/demo/export/parquet/' WITH (FORMAT="parquet");
+COPY DATABASE public TO '${SQLNESS_HOME}/demo/export/parquet/' WITH (FORMAT="parquet");
```
Comment (killme2008) [inline tests/cases/standalone/common/copy/copy_database_from_fs_parquet.result:9]: Why not use `data_home`? It's what it is.

Comment (v0y4g3r) [reply tests/cases/standalone/common/copy/copy_database_from_fs_parquet.result:9]: > Why not use `data_home`? It's what it is.

`data_home` is private to each datanode instance, while variable subsitution is carried out by sqlness runner, which is only aware of `sqlness_home`.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**